torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -1071,21 +1107,18 @@ def call_function(
             set_subgraph_inputs="flatten_manual",
         )
         cond_nn_modules = dict(tx.output.nn_modules)
-        if not isinstance(cond_r, TensorVariable):
-            unimplemented(
-                f"Expected cond_fn to return a tensor but got {cond_r.python_type()}",
-            )
-
-        # cond output checks
-        cond_r_meta = _extract_tensor_metadata(
-            cond_r.proxy.node.meta["example_value"], include_contiguity=False
-        )
-        if not cond_r_meta.dtype == torch.bool or not cond_r_meta.shape == torch.Size(
-            []
-        ):
-            unimplemented(
-                f"Expected cond_fn to return a tensor with shape (,) but got {cond_r_meta.shape}"
+        validate_subgraph_output_types(cond_r)
+        if isinstance(cond_r, TensorVariable):
+            cond_r_meta = _extract_tensor_metadata(
+                cond_r.proxy.node.meta["example_value"], include_contiguity=False
```

Comment:
Discussed offline: include_contiguity=False is fine to always do by default, because the memory format is determined by the strides.